common/modules/utils/src/lib.rs

```diff
@@ -68,6 +68,23 @@ pub trait UtilsModule {
     }
 
     fn merge_from_payments<T: FixedSupplyToken<Self::Api> + Mergeable<Self::Api> + TopDecode>(
+        &self,
+        payments: &mut PaymentsVec<Self::Api>,
+        mapper: &NonFungibleTokenMapper<Self::Api>,
+    ) -> T {
+        let first_payment = self.pop_first_payment(payments);
+        let base_attributes: T =
+            self.get_attributes_as_part_of_fixed_supply(&first_payment, &mapper);
+            mapper.nft_burn(first_payment.token_nonce, &first_payment.amount);
+
+        let output_attributes = self.merge_attributes_from_payments(base_attributes, &payments, &mapper);
+
+        self.burn_multi_esdt(&payments);
```

Comment:
Should not burn inside this method, as the caller might still need the payments. Let the caller handle the token burning.